subsys/net/ip/net_context.c

```diff
@@ -2401,9 +2401,29 @@ static int recv_udp(struct net_context *context,
 
 	ARG_UNUSED(timeout);
 
+	/* If the context already has a connection handler, it means it's
+	 * already registered. In that case, all we have to do is 1) update
+	 * the callback registered in the net_context and 2) update the
+	 * user_data using net_conn_change_callback() and 3) update the remote
+	 * address and port assuming connect() has been called.
+	 *
+	 * The callback function passed to net_conn_change_callback() must be
+	 * the same function as the one passed to net_conn_register(), not the
+	 * callback set for the net context passed by recv_udp.
+	 */
 	if (context->conn_handler) {
-		net_conn_unregister(context->conn_handler);
-		context->conn_handler = NULL;
+		context->recv_cb = cb;
+		ret = net_conn_change_callback(context->conn_handler,
+					       net_context_packet_received,
+					       user_data);
+		if (ret) {
+			goto end;
+		}
+		ret = net_conn_change_remote(context->conn_handler,
+					     context->flags & NET_CONTEXT_REMOTE_ADDR_SET ?
+								&context->remote : NULL,
+					     ntohs(net_sin(&context->remote)->sin_port));
```

Comment:
How about having `net_conn_update()` and do both.  Because we don't have any user of `net_conn_change_callback()` in tree and it's not a public API anyway.

So that a call-site will become something like:

```c
if (context->conn_handler) {
    ret = net_conn_update();
} else {
    ret = bind_default();
    if (ret >= 0) {
        ret = net_conn_register();
    }
}

return ret;
```

@jukkar any thoughts?

Reply:
I think having a static version of `net_conn_change_callback()` and `net_conn_change_remote()` in connection.c is fine. I assume that it's better to have `net_conn_update()` for net_context.c to consume.

Reply:
@yashi good point, I like your `net_conn_update()` idea, we could do it like that.

Reply:
Thank you for your review.
I will update the PR based on `net_conn_update()` idea.